### PR TITLE
V22F-697 Particpant control: gas, steer and brake

### PR DESCRIPTION
**Pull request by larsdekwant**
- Added checkbox for participant control: brake
- Reordered and renamed participant control checkboxes to match the blackbox control checkboxes
- Added functionality to the actual settings so the partipant can only control the specified controls.
- Fixed bug where particpant control on/off setting in controls menu was invisible.

NOTE: nothing changed to the actual steering/throttle/brake functionality, only that it is within an if condition now.

### CODE REVIEW

File: speed-dreams/source-2.2.3/src/modules/userinterface/textonly/textonly.cpp [hunk 1]
```diff
@@ -142,9 +142,9 @@ bool LoadReplayConfiguration(GfRaceManager*& p_selRaceMan) {
     SMediator::GetInstance()->SetMaxTime(maxTime);
 
     tParticipantControl participantControl{};
-    participantControl.ControlGas = StringToBool(GfParmGetStr(replaySettingsHandle, PATH_PARTICIPANT_CONTROL, KEY_PARTICIPANT_CONTROL_CONTROL_GAS, "false"));
+    participantControl.ControlAccel = StringToBool(GfParmGetStr(replaySettingsHandle, PATH_PARTICIPANT_CONTROL, KEY_PARTICIPANT_CONTROL_CONTROL_GAS, "false"));
     participantControl.ControlInterventionToggle = StringToBool(GfParmGetStr(replaySettingsHandle, PATH_PARTICIPANT_CONTROL, KEY_PARTICIPANT_CONTROL_CONTROL_INTERVENTION_TOGGLE, "false"));
-    participantControl.ControlSteering = StringToBool(GfParmGetStr(replaySettingsHandle, PATH_PARTICIPANT_CONTROL, KEY_PARTICIPANT_CONTROL_CONTROL_STEERING, "false"));
+    participantControl.ControlSteer = StringToBool(GfParmGetStr(replaySettingsHandle, PATH_PARTICIPANT_CONTROL, KEY_PARTICIPANT_CONTROL_CONTROL_STEERING, "false"));
```
Comment (suppergerrie2) [inline speed-dreams/source-2.2.3/src/modules/userinterface/textonly/textonly.cpp:147]: Missing ControlBrake

File: speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/backend/Recorder.cpp [hunk 2]
```diff
@@ -102,9 +102,9 @@ void Recorder::WriteRunSettings(const tCarElt* p_carElt, const tTrack* p_track,
     GfParmSetStr(settingsFileHandle, PATH_TRACK, KEY_CATEGORY, p_track->category);
     GfParmSetStr(settingsFileHandle, PATH_TRACK, KEY_NAME, p_track->name);
 
-    GfParmSetStr(settingsFileHandle, PATH_PARTICIPANT_CONTROL, KEY_PARTICIPANT_CONTROL_CONTROL_GAS, BoolToString(p_participantControl.ControlGas));
+    GfParmSetStr(settingsFileHandle, PATH_PARTICIPANT_CONTROL, KEY_PARTICIPANT_CONTROL_CONTROL_GAS, BoolToString(p_participantControl.ControlAccel));
     GfParmSetStr(settingsFileHandle, PATH_PARTICIPANT_CONTROL, KEY_PARTICIPANT_CONTROL_CONTROL_INTERVENTION_TOGGLE, BoolToString(p_participantControl.ControlInterventionToggle));
-    GfParmSetStr(settingsFileHandle, PATH_PARTICIPANT_CONTROL, KEY_PARTICIPANT_CONTROL_CONTROL_STEERING, BoolToString(p_participantControl.ControlSteering));
+    GfParmSetStr(settingsFileHandle, PATH_PARTICIPANT_CONTROL, KEY_PARTICIPANT_CONTROL_CONTROL_STEERING, BoolToString(p_participantControl.ControlSteer));
```
Comment (suppergerrie2) [inline speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/backend/Recorder.cpp:107]: Missing ControlBrake

File: speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/backendUnitTests/ConfigTests.cpp [hunk 3]
```diff
@@ -73,8 +73,8 @@ void PControlTest(bool p_intervention, bool p_gas, bool p_steer, bool p_force)
     config.SetPControlSettings(arr);
     tParticipantControl pControl = config.GetPControlSettings();
     ASSERT_EQ(arr.ControlInterventionToggle, pControl.ControlInterventionToggle);
-    ASSERT_EQ(arr.ControlSteering, pControl.ControlSteering);
-    ASSERT_EQ(arr.ControlGas, pControl.ControlGas);
+    ASSERT_EQ(arr.ControlSteer, pControl.ControlSteer);
+    ASSERT_EQ(arr.ControlAccel, pControl.ControlAccel);
```
Comment (suppergerrie2) [inline speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/backendUnitTests/ConfigTests.cpp:77]: Missing ControlBrake

File: speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/backendUnitTests/MediatorTests.cpp [hunk 4]
```diff
@@ -164,8 +164,8 @@ void PControlTestMediator(bool p_intervention, bool p_gas, bool p_steer, bool p_
     SDAConfigMediator::GetInstance()->SetPControlSettings(arr);
     tParticipantControl pControl = SDAConfigMediator::GetInstance()->GetPControlSettings();
     ASSERT_EQ(arr.ControlInterventionToggle, pControl.ControlInterventionToggle);
-    ASSERT_EQ(arr.ControlSteering, pControl.ControlSteering);
-    ASSERT_EQ(arr.ControlGas, pControl.ControlGas);
+    ASSERT_EQ(arr.ControlSteer, pControl.ControlSteer);
+    ASSERT_EQ(arr.ControlAccel, pControl.ControlAccel);
```
Comment (suppergerrie2) [inline speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/backendUnitTests/MediatorTests.cpp:168]: Missing ControlBrake

File: speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/backendUnitTests/RecorderTests.cpp [hunk 5]
```diff
@@ -337,8 +337,8 @@ TEST(RecorderTests, WriteRunSettingsTests)
 
     tParticipantControl participantControl;
     participantControl.ControlInterventionToggle = random.NextBool();
-    participantControl.ControlGas = random.NextBool();
-    participantControl.ControlSteering = random.NextBool();
+    participantControl.ControlAccel = random.NextBool();
+    participantControl.ControlSteer = random.NextBool();
```
Comment (suppergerrie2) [inline speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/backendUnitTests/RecorderTests.cpp:341]: Missing ControlBrake

**Review by suppergerrie2 — CHANGES_REQUESTED** [review]

File: speed-dreams/source-2.2.3/data/data/menu/controlconfigmenu.xml [hunk 6]
```diff
@@ -438,7 +438,7 @@
     </section>
 
     <!--SIMULATED DRIVING ASSISTANCE-->
-    <section name="intervention toggle button">
+    <section name="intervention-toggle button">
         <attstr name="type" val="text button"/>
         <attstr name="show box" val="no"/>
         <attstr name="text" val="MOUSE_MIDDLE_BUTTON"/>
```
Comment (sjoeman1) [inline speed-dreams/source-2.2.3/data/data/menu/controlconfigmenu.xml:444]: the intervention toggle can't be a key on the keyboard, could you add that to the hint?

**Review by sjoeman1 — CHANGES_REQUESTED** [review]
The intervention on/off can't use keyboard buttons, could you add that to the hint?
The intervention on/off does not appear to work correctly, also test with abandon race as well as quiting the game

File: speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/backend/Recorder.cpp [hunk 7]
```diff
@@ -343,41 +388,20 @@ bool Recorder::ValidateAndUpdateRecording(const filesystem::path& p_recordingFol
...
-        UpdateV1RecorderToV2(settingsHandle);
-        version++;
-    }
-
-    // Update version 2 to version 3 recording
-    if (version == 2)
-    {
-        UpdateV2RecorderToV3(settingsHandle);
-        version++;
+        // If upgrading fails, release handle and exit loop.
```
Comment (suppergerrie2) [inline speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/backend/Recorder.cpp:396]: This does more than exiting the loop, it returns the entire function

File: speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/backendUnitTests/RecorderTests.cpp [hunk 8]
```diff
@@ -439,102 +440,90 @@ void TestV0ToV1Changes(void* p_upgradedRunSettingsHandle, filesystem::path& p_to
...
 }
 
+/// @brief TestFixture to help with testing the recording upgrades in a systematic way.
+class RecorderUpgradeVersionTestFixture : public ::testing::TestWithParam<int>
+{
+};
 
-/// @brief Test whether the upgrade from V2 is done correctly
-TEST(RecorderTests, UpgradeFromV2Test)
+/// @brief Tests whether the base-recording is succesfully upgrade to the target version.
```
Comment (suppergerrie2) [inline speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/backendUnitTests/RecorderTests.cpp:494]:
```suggestion
/// @brief Tests whether the base-recording is successfully upgrade to the target version.
```

File: speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/backendUnitTests/RecorderTests.cpp [hunk 9]
```diff
@@ -439,102 +440,90 @@ void TestV0ToV1Changes(void* p_upgradedRunSettingsHandle, filesystem::path& p_to
...
-    INIT_VALIDATE_OR_UPGRADE_TEST("v3-recording", toUpgrade);
-
-    ASSERT_TRUE(Recorder::ValidateAndUpdateRecording(toUpgrade));
-
-    ASSERT_TRUE(filesystem::exists(filesystem::path(toUpgrade).append(USER_INPUT_RECORDING_FILE_NAME)));
-    ASSERT_TRUE(filesystem::exists(filesystem::path(toUpgrade).append(SIMULATION_DATA_RECORDING_FILE_NAME)));
-    ASSERT_TRUE(filesystem::exists(filesystem::path(toUpgrade).append(DECISIONS_RECORDING_FILE_NAME)));
+/// @brief       Generates all the tests for the recording upgrades
+/// @param Range The versions to upgrade the recordings to, is a range from [1..RECORDER_VERSION]
+INSTANTIATE_TEST_CASE_P(
```
Comment (suppergerrie2) [inline speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/backendUnitTests/RecorderTests.cpp:516]: INSTANTIATE_TEST_CASE_P seems to be deprecated. It says to use INSTANTIATE_TEST_SUITE_P

**Review by suppergerrie2 — CHANGES_REQUESTED** [review]
Add changes to Code Changes document

**Review by sjoeman1 — APPROVED** [review]
The intervention on/off appears to work right now, I did delete my settings folder this time to test it properly, so maybe that had something to do with it

**Review by suppergerrie2 — APPROVED** [review]

**Pull request merged**